pkg/core/util/repository/azrcr.go

```diff
@@ -180,6 +174,167 @@ func GetImageAndShaFromDownload(driverConfig *eUtils.DriverConfig, pluginToolCon
 }
 
 // Pushes image to docker registry from: "rawImageFile", and "pluginname" in the map pluginToolConfig.
+// https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry#readme-examples
 func PushImage(driverConfig *eUtils.DriverConfig, pluginToolConfig map[string]interface{}) error {
-	return errors.New("Not defined")
+	err := ValidateRepository(driverConfig, pluginToolConfig)
+
+	if err != nil {
+		return err
+	}
+
+	svc, err := azidentity.NewClientSecretCredential(
+		pluginToolConfig["azureTenantId"].(string),
+		pluginToolConfig["azureClientId"].(string),
+		pluginToolConfig["azureClientSecret"].(string),
+		nil)
+
+	if err != nil {
+		return err
+	}
+
+	azrClient, err := azcontainerregistry.NewClient(
+		pluginToolConfig["acrrepository"].(string),
```

Comment:
Same... it should be a check at top that all keys are available before we proceed.

Reply:
Already covered